code-review/examples/learning_culture.html.erb

```diff
@@ -0,0 +1,11 @@
+
+<% @posts = Post.all %>
+
+<% @posts.each do |post| %>
+  <h2><%= post.title %></h2>
+  <ul>
+    <% post.comments.each do |comment| %>
+      <li><%= comment.body %></li>
+    <% end %>
+  </ul>
+<% end %>
```

Comment:
# Promote a learning culture
When doing CR we can try to, at the same time we are trying to have the best base of code, to teach if it's the case. In this example, this code has a N+1, and beyond to request for a change, we can explain why is this happening promoting to creation of knowledge between our pairs.

```
In this code, you first retrieve all posts using Post.all, and then, for each post, you retrieve its associated comments with post.comments. This results in an N+1 query problem because you make an additional query for comments for each post.

We can try this way that eliminates the N+1 problem 

# Fetch all posts and include comments
@posts = Post.includes(:comments).all

# In your view, you can display posts and comments without causing additional queries
<% @posts.each do |post| %>
  <h2><%= post.title %></h2>
  <ul>
    <% post.comments.each do |comment| %>
      <li><%= comment.body %></li>
    <% end %>
  </ul>
<% end %>
```